.template/addons/github/.github/workflows/test.yml.tt

```diff
@@ -0,0 +1,93 @@
+name: Test
+
+on: push
+
+env:
+  DOCKER_REGISTRY_HOST: ${{ secrets.DOCKER_REGISTRY_HOST }}
+  DOCKER_REGISTRY_USERNAME: ${{ github.repository_owner }}
+  DOCKER_REGISTRY_TOKEN: ${{ secrets.DOCKER_REGISTRY_TOKEN }}
+  DOCKER_IMAGE: ${{ github.repository }}
+
+  # Set the default docker-compose file
+  COMPOSE_FILE: docker-compose.test.yml
+
+jobs:
+  build:
+    name: Build docker image
+    runs-on: ubuntu-latest
+    steps:
+      - name: Cancel previous runs
+        uses: styfle/cancel-workflow-action@0.6.0
+        with:
+          access_token: ${{ github.token }}
+
+      - uses: actions/checkout@v2.3.4
+
+      - name: Set env BRANCH_TAG
+        uses: nimblehq/branch-tag-action@v1.2
+
+      - name: Login to docker registry
+        uses: docker/login-action@v1.6.0
+        with:
+          registry: ${{ env.DOCKER_REGISTRY_HOST }}
+          username: ${{ env.DOCKER_REGISTRY_USERNAME }}
+          password: ${{ env.DOCKER_REGISTRY_TOKEN }}
+
+      - name: Pull docker image
+        if: ${{ env.BRANCH_TAG != 'latest' && env.BRANCH_TAG != 'development' }}
+        run: docker-compose pull test || true
+
+      - name: Build docker image
+        run: |
+          bin/docker-prepare
+          docker-compose build
+
+      - name: Push docker image
+        run: |
+          docker-compose push test
+
+  unit_tests:
```

Comment:
https://github.com/nimblehq/rails-templates/pull/225#discussion_r552410662

I moved them to different jobs because I want them to run parallel.
Let say we have a lot of test cases later, so it will take time to run. So I create 3 jobs, the first one to build the docker image, the second and the third will use that image to run unit test and system test in parallel, so it will reduce the time.